alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -73,81 +78,66 @@ class DocLevelMonitorQueries(private val client: AdminClient, private val cluste
         val clusterState = clusterService.state()
 
         val getAliasesRequest = GetAliasesRequest(index)
-        queryClient.admin().indices().getAliases(
-            getAliasesRequest,
-            object : ActionListener<GetAliasesResponse> {
-                override fun onResponse(getAliasesResponse: GetAliasesResponse?) {
-                    val aliasIndices = getAliasesResponse?.aliases?.keys()?.map { it.value }
-                    val isAlias = aliasIndices != null && aliasIndices.isNotEmpty()
-                    val indices = if (isAlias) aliasIndices else listOf(index)
-                    val indexRequests = mutableListOf<IndexRequest>()
-                    log.info("indices: $indices")
-                    indices?.forEach { indexName ->
-                        if (clusterState.routingTable.hasIndex(indexName)) {
-                            val indexMetadata = clusterState.metadata.index(indexName)
-                            if (indexMetadata.mapping()?.sourceAsMap?.get("properties") != null) {
-                                val properties = (
-                                    (indexMetadata.mapping()?.sourceAsMap?.get("properties"))
-                                        as Map<String, Map<String, Any>>
-                                    )
-                                val updatedProperties = properties.entries.associate {
-                                    "${it.key}_${indexName}_$monitorId" to it.value
-                                }.toMutableMap()
-
-                                val updateMappingRequest = PutMappingRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
-                                updateMappingRequest.source(mapOf<String, Any>("properties" to updatedProperties))
+        val getAliasesResponse: GetAliasesResponse = queryClient.suspendUntil {
+            queryClient.admin().indices().getAliases(getAliasesRequest, it)
+        }
+        val aliasIndices = getAliasesResponse.aliases?.keys()?.map { it.value }
+        val isAlias = aliasIndices != null && aliasIndices.isNotEmpty()
+        val indices = if (isAlias) aliasIndices else listOf(index)
 
-                                queryClient.admin().indices().putMapping(
-                                    updateMappingRequest,
-                                    object : ActionListener<AcknowledgedResponse> {
-                                        override fun onResponse(response: AcknowledgedResponse) {
+        indices?.forEach { indexName ->
+            if (clusterState.routingTable.hasIndex(indexName)) {
```

Comment:
Is this is what you were proposing to execute in parallel? Seems like a good candidate!

Reply:
Yea this would be a great candidate to parallelize